### ISSUE TITLE
Math not rendering in parameter hints

### ISSUE THREAD

**Issue by ibobak** [jupyterlab/jupyterlab#4403]
When I press Ctrl+Tab, I am getting help on parameters. Here is a piece of code from pymc3 - http://docs.pymc.io/notebooks/getting_started

![2018-04-17_11-22-20](https://user-images.githubusercontent.com/8807043/38857411-e2fdfe22-4231-11e8-838e-292358c24203.jpg)

**Comment by ibobak** [comment]
Sorry - I meant "shift tab" on the screenshot.

**Comment by jasongrout** [comment]
We talked about this at last scipy. If I recall correctly, @SylvainCorlay mentioned that @ccordoba12 had a library that did something nice with formatting docstrings.

Several people from the Sage community have also expressed interest in having docstring render math (can't remember who, but I'll CC @williamstein and @jdemeyer).

**Comment by jasongrout** [comment]
Yes, that's right. docrepr was what I was thinking of.

**Comment by ccordoba12** [comment]
Yep, that's it. Unfortunately we haven't had time to integrate it with Spyder.

Jason, perhaps we could work together (I mean, JLab and Spyder) to make this library work for both projects?

@SylvainCorlay already added the necessary pieces to IPython to use it, but we'd need to update it and make it configurable so that both projects can customize it according to their needs.

**Comment by joelostblom** [comment]
This would be really slick to have and it was recently raised again in #8581 (and previously in #5124).

It seems like docrepr is now integrated in Spyder which has [rich rendering in its help panel](https://docs.spyder-ide.org/current/help.html). It also seems like [this was/is working in the classic notebook](https://twitter.com/SylvainCorlay/status/926206993311961088), but I could not get the same trick to work in jupyterlab currently (2.2.8).

I would be happy to try to help with this if you think it is a good first issue, but I also don't have any idea upfront of what would be required for docrepr to work with JupyterLab. @ccordoba12 or @jasongrout do you know roughly what would be required to make this integration work?

**Comment by dhirschfeld** [comment]
Seems somewhat related to #9663?

**Comment by krassowski** [comment]
> I believe that the solution to the rst rendering issues is to convert those to markdown at backend (this is kernel for <kbd>shitf</kbd> + <kbd>tab</kbd> and LSP server such as pyls or jedi-language-server for the LSP extension completer suggestions). This can be enabled by pandoc. The rationale is that rst is a very Python-centric format and Jupyter as a project aims to be inclusive of many more languages and markdown seems like a reasonable common ground for documentation. Also, the current JupyterLab renderers are well capable of rendering any markdown including latex equations.

_Originally posted by @krassowski in https://github.com/jupyterlab/jupyterlab/issues/9663#issuecomment-773490474_

Related to https://github.com/palantir/python-language-server/issues/836
And to: https://github.com/pappasam/jedi-language-server/issues/22

I would be happy to work upstream on IPython front and language servers front to convert docstrings to proper markdown. There are multiple solutions as discussed in https://github.com/pappasam/jedi-language-server/issues/22#issuecomment-633257107 and some more including the use of:

- [napoleon](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html) - supports NumPy and Google in addition to RST
- [pydoc-markdown](https://pypi.org/project/pydoc-markdown/) - supports Sphinx, Google, Pydoc-Markdown

[Pyment](https://github.com/dadadel/pyment) sounded most promising and I am sad to read that it did not work out when @pappasam tried it back in May. @pappasam how severe were the limitations? (sorry for pinging you on another repo, but I believe the conversation above might be of your interest too).

The third-party converter would of course be an **optional** dependency of the kernel (and optional or not optional for language servers depending what is the maintainers will).

**Comment by kdonbekci** [comment]
Any updates on this? Also, related to math in viewing docstring as output of a cell when invoking `foo?`

**Comment by joelostblom** [comment]
It seems like this is now possible via `docrepr` as detailed in this blog post! https://blog.jupyter.org/inspector-jupyterlab-404cce3e1df6
